### PR TITLE
Remove unnecessary creations of lists at all

### PR DESCRIPTION
**Pull request by horpto**

### CODE REVIEW

File: gensim/models/ldamodel.py [hunk 1]
```diff
@@ -567,15 +567,18 @@ def init_dir_prior(self, prior, name):
         if isinstance(prior, six.string_types):
             if prior == 'symmetric':
                 logger.info("using symmetric %s at %s", name, 1.0 / self.num_topics)
-                init_prior = np.asarray([1.0 / self.num_topics for i in xrange(prior_shape)], dtype=self.dtype)
+                init_prior = np.fromiter((1.0 / self.num_topics for i in xrange(prior_shape)),
+                                            dtype=self.dtype, count=prior_shape)
```
Comment (piskvorky) [inline gensim/models/ldamodel.py]: Hanging indent please, here and elsewhere.

We avoid vertical indent because it leads to inconsistent visual indenting (like here), especially when the lines are edited in the future.

File: gensim/test/test_fasttext.py [hunk 2]
```diff
@@ -674,7 +674,6 @@ def online_sanity(self, model):
                 terro.append(l)
             else:
                 others.append(l)
-        self.assertTrue(all(['terrorism' not in l for l in others]))
```
Comment (piskvorky) [inline gensim/test/test_fasttext.py]: Why this change?

**Comment by menshikh-iv** [comment]
ping @horpto, please fix @piskvorky comments and I'll merge current PR (don't forget to merge fresh `develop` first)

File: gensim/test/test_fasttext.py (hunk 2 above)
Comment (horpto) [reply gensim/test/test_fasttext.py]: I thought that this line does not check anything worthwhile. By code it is clear that this check is always true. It's not critical for me, so I return this line back.

File: gensim/models/ldamodel.py (hunk 1 above)
Comment (horpto) [reply gensim/models/ldamodel.py]: Done.

File: gensim/models/ldamodel.py [hunk 3]
```diff
@@ -567,15 +567,18 @@ def init_dir_prior(self, prior, name):
         if isinstance(prior, six.string_types):
             if prior == 'symmetric':
                 logger.info("using symmetric %s at %s", name, 1.0 / self.num_topics)
-                init_prior = np.asarray([1.0 / self.num_topics for i in xrange(prior_shape)], dtype=self.dtype)
+                init_prior = np.fromiter((1.0 / self.num_topics for i in xrange(prior_shape)),
+                                         dtype=self.dtype, count=prior_shape)
```
Comment (piskvorky) [inline gensim/models/ldamodel.py]: This is still vertical indent.

In gensim, we use hanging indent always, consistently: https://www.python.org/dev/peps/pep-0008/#id3
